civis/tests/test_io.py

```diff
@@ -383,25 +389,37 @@ def _dump_json(file_id, buf, *args, **kwargs):
     assert out == obj
 
 
-@mock.patch('civis.io._files._civis_to_file')
-@mock.patch('%s.open' % __name__, create=True)
-def test_civis_to_file_local(mock_open, mock_civis_to_file_helper):
+@mock.patch('civis.io._files._civis_to_file', autospec=True)
+def test_civis_to_file_local(mock_civis_to_file_helper):
     # Test that passing a path to civis_to_file opens a file.
-    civis.io.civis_to_file(123, "foo")
-    mock_open.return_value = mock_file = mock.Mock()
-    assert mock_open.called_once_with("foo", "wb")
-    assert mock_civis_to_file_helper.called_once_with(123, mock_file)
-
-
-@mock.patch('civis.io._files._file_to_civis')
-@mock.patch('%s.open' % __name__, create=True)
-def test_file_to_civis(mock_open, mock_file_to_civis_helper):
+    def mock_write(file_id, buf, **kwargs):
+        buf.write(str(file_id).encode())
+    mock_civis_to_file_helper.side_effect = mock_write
+
+    test_obj = '123'
+    with TemporaryDirectory() as tdir:
+        fname = os.path.join(tdir, "foo")
+        assert not os.path.exists(fname)
+        civis.io.civis_to_file(int(test_obj), fname)
+        assert os.path.exists(fname)
+        with open(fname, 'rt') as fin:
+            assert fin.read() == test_obj
+
+
+@mock.patch('civis.io._files._file_to_civis', autospec=True)
+def test_file_to_civis(mock_file_to_civis_helper):
     # Test that passing a path to file_to_civis opens a file.
-    civis.io.file_to_civis("foo", "foo_name")
-    mock_open.return_value = mock_file = mock.Mock()
-    assert mock_open.called_once_with("foo", "rb")
-    assert mock_file_to_civis_helper.called_once_with(
-        "foo", "foo_name", mock_file)
+    def mock_read(buf, name, **kwargs):
+        return buf.read()
+    mock_file_to_civis_helper.side_effect = mock_read
```

Comment:
What was the issue with the previous incarnation of this test mocking open?

Reply:
The mock wasn't being called. Rather than trying to figure out where to mock, it seemed simpler and more robust to implementation changes if I used real files.